crates/tokio-util/src/event_listeners.rs

```diff
@@ -1,46 +1,38 @@
-use tokio::sync::mpsc;
-use tokio_stream::wrappers::UnboundedReceiverStream;
+use tokio::sync::broadcast::{self, Sender};
+use tokio_stream::wrappers::BroadcastStream;
+use tracing::error;
 
-/// A collection of event listeners for a task.
-#[derive(Clone, Debug)]
+const DEFAULT_SIZE_BROADCAST_CHANNEL: usize = 1000;
+
+/// A bounded broadcast channel for a task.
+#[derive(Debug, Clone)]
 pub struct EventListeners<T> {
-    /// All listeners for events
-    listeners: Vec<mpsc::UnboundedSender<T>>,
+    /// The sender part of the broadcast channel
+    sender: Sender<T>,
 }
 
-impl<T> Default for EventListeners<T> {
+impl<T: Clone + Send + Sync + 'static> Default for EventListeners<T> {
```

Comment:
```suggestion
impl<T> Default for EventListeners<T> where T: Clone + Send + Sync + 'static {
```
we try to stick to this syntax

Reply:
thanks!